### ISSUE TITLE
自训练模型报错，及单机多卡配置后默认单卡

### ISSUE THREAD

**Issue by alittejoke** [PaddlePaddle/PaddleOCR#3410]
训练机paddlepaddle-gpu-2.02post110 双3060
测试机paddlepaddle-gpu-2.02post110 双3060
测试机paddlepaddle-gpu-1.85post107 单1650s
1.
![image](https://user-images.githubusercontent.com/65384545/126733329-77cc0867-ef12-4715-878c-b7a9a1ff6b89.png)
训练后转为inference模型是如何与
![image](https://user-images.githubusercontent.com/65384545/126733417-be1d7d46-0b7d-40d2-9a46-1080cd8156e0.png)
这个里面的models与params匹配的 直接修改吗？

2.加载过程中报错
![16270106691295](https://user-images.githubusercontent.com/65384545/126734267-c9adb8ed-1dbe-4889-b405-2321da65f01d.png)

3.如何设置单机多卡训练 
单机单卡能够正常进行训练
windows系统 
![16270109291103](https://user-images.githubusercontent.com/65384545/126734723-0cb7e235-bfc5-486c-8c70-bef72872beb5.png)
我的windows 所以改为 os.system(set CUDA_VISIBLE_DEVICES=0,1)
我也是设置之后发现了同样的问题 训练还是只调用一张卡

同时尝试修改了train.py里边的dist的判断条件默认为dist
后来又尝试了   
# 多卡并行计算
    dist.spawn(train)
但还是device=0 另一张卡根本没有调用 

4. 训练参数是修改configs\rec\ch_ppocr_v2.0\rec_chinese_common_train_v2.0.yml
相关配置为
Global:
  use_gpu: true
  epoch_num: 300
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./output/rec_chinese_common_v2.0
  save_epoch_step: 3
  # evaluation is run every 5000 iterations after the 4000th iteration
  eval_batch_step: [0, 2000]
  cal_metric_during_train: True
  pretrained_model: D:\PaddleOCR\pretrain_models\rec_r34_vd_none_bilstm_ctc_v2.0_train/best_accuracy
  checkpoints:
  save_inference_dir: D:\PaddleOCR\pretrain_models\inference
  use_visualdl: False
  infer_img: doc/imgs_words/ch/word_1.jpg
  # for data or label process
  character_dict_path: D:\PaddleOCR\ppocr/utils/ppocr_keys_v1.txt
  character_type: ch
  max_text_length: 25
  infer_mode: False
  use_space_char: True
  save_res_path: ./output/rec/predicts_chinese_r34common_v2.0.txt

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.0005
  regularizer:
    name: 'L2'
    factor: 0.00002

Architecture:
  model_type: rec
  algorithm: CRNN
  Transform:
  Backbone:
    name: ResNet
    layers: 34
  Neck:
    name: SequenceEncoder
    encoder_type: rnn
    hidden_size: 256
  Head:
    name: CTCHead
    fc_decay: 0.00002

Loss:
  name: CTCLoss

PostProcess:
  name: CTCLabelDecode

Metric:
  name: RecMetric
  main_indicator: acc

Train:
  dataset:
    name: PGDataSet
    data_dir: D:/PaddleOCR/train_data/train_img/
    label_file_list: ["D:/PaddleOCR/train_data/train_img/rec_gt.txt"]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - RecAug: 
      - CTCLabelEncode: # Class handling label
      - RecResizeImg:
          image_shape: [3, 32, 320]
      - KeepKeys:
          keep_keys: ['image', 'label', 'length'] # dataloader will return list in this order
  loader:
    shuffle: True
    batch_size_per_card: 80
    drop_last: True
    num_workers: 2

Eval:
  dataset:
    name: PGDataSet
    data_dir: D:/PaddleOCR/train_data/test_img/
    label_file_list: ["D:/PaddleOCR/train_data/test_img/rec_gt.txt"]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - CTCLabelEncode: # Class handling label
      - RecResizeImg:
          image_shape: [3, 32, 320]
      - KeepKeys:
          keep_keys: ['image', 'label', 'length'] # dataloader will return list in this order
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 80
    num_workers: 2

6.期待请尽快明确的回复，多谢！！！

**Comment by littletomatodonkey** [comment]
1. 安装下paddle2.0.2吧，你的paddle看着是1.8.5的，有点旧了
2. windows目前应该还不支持多卡运行(多卡依赖于nccl)，或者你可以试下distributed.launch启动下，可以参考train.sh脚本

**Comment by alittejoke** [comment]
已经安装2.0.2 问题没有解决 测试机里就有2.0.2环境

**Comment by alittejoke** [comment]
distributed.launch 启动失败 原因是windows不支持nccl

**Comment by littletomatodonkey** [comment]
嗯，那就只能单卡了

**Comment by littletomatodonkey** [comment]
嗯嗯，那目前只能单卡在windows上跑，另外，看你的报错，用的还是1.8.5的paddle，建议再看下python环境是否正确哈

**Comment by alittejoke** [comment]
![image](https://user-images.githubusercontent.com/65384545/126765709-211feb85-c92f-499b-8fb6-96b0d16c4147.png)
1.inference模型如何转为params与model模型？
2.是否直接替换 paddleocr 1.0.1.whl 安装后 site_packages下的paddleocr\tools\infer\utility.py 
   替换为PaddleOCR训练项目 下的 PaddleOCR\tools\infer\utility.py

**Comment by alittejoke** [comment]
我看到两个中的读模型的部分不同

**Comment by alittejoke** [comment]
我更换版本1.85为2.02后报错消失了 ，但没有检测结果，结果显示为
![16270346354158](https://user-images.githubusercontent.com/65384545/126767336-36487c65-d289-4e11-ad0b-c611f2e3ea5a.png)

**Comment by alittejoke** [comment]
![企业微信截图_162713565170](https://user-images.githubusercontent.com/65384545/126871086-2969e467-fe4c-4673-814e-bb318285f52e.png)
paddlepaddle-gpu1.8.5post107     paddleocr版本1.0.1
第一张图片是两张图使用model，params模型能够正常运行
![无标题](https://user-images.githubusercontent.com/65384545/126871123-a43006b1-548f-4a14-9a03-22c1b1ad81f8.png)
paddlepaddle-gpu2.0.2post110    paddleocr版本2.0.2
第二张图片是同样的2张图使用rec-inference-server模型报错

请问第二张图paddlepaddle-gpu2.0.2post110报错原因
第二个问题 inference模型如何转换为model,params模型
